03.bowling/bowling.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+class BowlingScoreCalculator
+  def initialize(scores_str)
+    @scores_str = scores_str
+  end
+
+  def print_total_score
+    scores_by_bowl = split_scores_by_bowl(scores_str: @scores_str)
+    scores_by_frame = group_scores_by_frame(scores_by_bowl:)
+    puts calculate_total_score(scores_by_frame:)
+  end
+
+  private
+
+  def split_scores_by_bowl(scores_str:)
+    scores_str.gsub('X', '10').split(',').map(&:to_i)
+  end
+
+  def group_scores_by_frame(scores_by_bowl:)
+    frame = []
+    game = []
+
+    scores_by_bowl.each do |score|
+      frame << score
+      next unless score == 10 || frame.size == 2
+      next if game.size == 9
+
+      game << frame
+      frame = []
+    end
+    game << frame
```

Comment:
`<<` は要素の追加がメインで、戻り値が返ってくるのは副作用みたいなものです。
これだけ見ると「要素を追加して終わり」と読めてしまうので、以下のようにして明示的に戻り値を返した方がわかりやすいですね。

```suggestion
    game << frame # 要素の追加
    game # 戻り値を返す
```

なお、`<<` の戻り値は https://docs.ruby-lang.org/ja/latest/method/Array/i/=3c=3c.html に書かれているようなコードを書くために使われることが多いです。（こういうコードも滅多に書きませんが。。）

```ruby
ary = [1]
ary << 2 << 3 << 4
p ary   #=> [1, 2, 3, 4]
```